### PR TITLE
fix upload android mapping.txt

### LINKED ISSUE #54: Android Mapping.txt
Hello,

in the HockeyApp action we used the "dysm" property to upload the mapping.txt.
This does not work with the AppCenter-Plugin.. See the screenshot.
mapping_files

What is the right way, to upload it?

Thanks
Michael

### PR DESCRIPTION
**Pull request by enizbilgin**
Closes #54  

Screen_Shot_2019-04-26_at_20_50_52

### CODE REVIEW

**Comment by evkhramkov** [comment]
@enizbilgin thanks a lot for your contribution!
One small request - could you please leave only issue related changes (new helper method and action logic)? If you'd like to suggest code style changes, it would be better to have them in a separate PR.

File: lib/fastlane/plugin/appcenter/helper/appcenter_helper.rb [hunk 1]
```diff
@@ -33,14 +33,50 @@ def self.create_release_upload(api_token, owner_name, app_name)
...
+      def self.create_mapping_upload(api_token, owner_name, app_name, build_number, version)
+        connection = self.connection
+
+        response = connection.post do |req|
+          req.url("/v0.1/apps/#{owner_name}/#{app_name}/symbol_uploads")
+          req.headers["X-API-Token"] = api_token
+          req.headers["internal-request-source"] = "fastlane"
+          req.body = {
+            symbol_type: "AndroidProguard",
+            file_name: "upload-mapping.txt",
```
Comment (evkhramkov) [inline lib/fastlane/plugin/appcenter/helper/appcenter_helper.rb]: let's use actual file name here

Comment (enizbilgin) [reply lib/fastlane/plugin/appcenter/helper/appcenter_helper.rb]: I properly fixed merge request

File: lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb [hunk 2]
```diff
@@ -289,7 +297,19 @@ def self.available_options
                                   env_name: "APPCENTER_DISTRIBUTE_RELEASE_NOTES_LINK",
                                description: "Additional release notes link",
                                   optional: true,
-                                      type: String)
+                                      type: String),
+
+          FastlaneCore::ConfigItem.new(key: :build_number,
+                                       env_name: "APPCENTER_DISTRIBUTE_BUILD_NUMBER",
+                                       description: "Android apk build number",
```
Comment (DennisPan) [inline lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb]: I would suggest change the description here to: `The build number. Used (and required) for uploading Android ProGuard mapping file.`

File: lib/fastlane/plugin/appcenter/helper/appcenter_helper.rb [hunk 3]
```diff
@@ -54,6 +54,42 @@ def self.create_release_upload(api_token, owner_name, app_name)
         end
       end
 
+      # creates new mapping upload in appcenter
+      # returns:
+      # symbol_upload_id
+      # upload_url
+      # expiration_date
+      def self.create_mapping_upload(api_token, owner_name, app_name, file_name,build_number, version)
```
Comment (DennisPan) [inline lib/fastlane/plugin/appcenter/helper/appcenter_helper.rb]: nit: missing space before `build_number`

File: lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb [hunk 4]
```diff
@@ -289,7 +297,19 @@ def self.available_options
...
+                                       optional: true,
+                                       type: String),
+
+          FastlaneCore::ConfigItem.new(key: :version,
+                                       env_name: "APPCENTER_DISTRIBUTE_VERSION",
+                                       description: "Android apk version name",
```
Comment (DennisPan) [inline lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb]: Same here. Would suggest change this to `The version number. Used (and required) for uploading Android ProGuard mapping file.`

**Pull request merged**